propagation/trace_context.go

```diff
@@ -59,12 +60,27 @@ func (tc TraceContext) Inject(ctx context.Context, carrier TextMapCarrier) {
 	// Clear all flags other than the trace-context supported sampling bit.
 	flags := sc.TraceFlags() & trace.FlagsSampled
 
-	h := fmt.Sprintf("%.2x-%s-%s-%s",
-		supportedVersion,
-		sc.TraceID(),
-		sc.SpanID(),
-		flags)
-	carrier.Set(traceparentHeader, h)
+	var sb strings.Builder
+	sb.Grow(2 + 32 + 16 + 2 + 3)
+	_, _ = sb.WriteString(versionPart)
+	traceID := sc.TraceID()
+	spanID := sc.SpanID()
+	flagByte := [1]byte{byte(flags)}
+	writeTraceParent(&sb, traceID[:], spanID[:], flagByte[:])
+	carrier.Set(traceparentHeader, sb.String())
+}
+
+func writeTraceParent(sb *strings.Builder, srcs ...[]byte) {
+	for _, src := range srcs {
+		_ = sb.WriteByte(parentDelimiter[0])
+		writeHex(sb, src)
+	}
+}
+
+func writeHex(sb *strings.Builder, src []byte) {
+	var dst [32]byte
```

Comment:
Using an array of this size is only valid because we know it will at most be passed a trace ID from above. Having this separate from `Inject`, which satisfies this assumption, seems error prone. What about declaring this function inline in `Inject` and flattening the `writeTraceParent` for loop there as well?

Reply:
you are right